linera-execution/src/system.rs

```diff
@@ -623,28 +622,27 @@ where
 
     pub async fn transfer(
         &mut self,
-        authenticated_signer: Option<Owner>,
-        authenticated_application_id: Option<UserApplicationId>,
+        authenticated_signer: Option<AccountOwner>,
+        authenticated_application_id: Option<ApplicationId>,
         source: AccountOwner,
         recipient: Recipient,
         amount: Amount,
     ) -> Result<Option<OutgoingMessage>, ExecutionError> {
-        match (source, authenticated_signer, authenticated_application_id) {
-            (AccountOwner::User(owner), Some(signer), _) => ensure!(
-                signer == owner,
+        if source == AccountOwner::chain() {
+            ensure!(
+                authenticated_signer.is_some()
+                    && self
+                        .ownership
+                        .get()
+                        .verify_owner(&authenticated_signer.unwrap()),
```

Comment:
Compare to https://github.com/linera-io/linera-protocol/pull/3593/files#r2009030749